### DISCUSSION TITLE
How do we add more Blender Functionality to Kubric Core?

### DISCUSSION THREAD

**Discussion by MrXandbadas** [google-research/kubric#193] [Q&A] [answered]
Is there a way to re-compile Kubric Core? As when I edit and make changes they aren't reflected/accessible from the examples or any test code I try.

Maybe I'm just missing a file where a reference needs to be included?
Edit: Maybe I'm just dumb. Please hold :P
XD

**Comment by taiya** [comment, marked as answer]
Hi @MrXandbadas if you want to modify the core, you should use the `kubruntudev` image.
This loads kubric from the current folder.

this is what I do when I need to modify the core (edits to library will be picked up live):

```
git clone git clone git@github.com:google-research/kubric.git
cd kubric
docker pull kubricdockerhub/kubruntudev
make examples/shapenet
```

And this happens because kubruntudev has this env variable:
```
ENV PYTHONPATH="/workspace"
```

And as you can see the `Makefile` mounts the content of the kubric download directory to `/workspace`.
Feel free to ask if you have further questions ;)


p.s. the makefile is really just a collection of shell files, it's just my way of not having 1000x one-line `script.sh` lying around

**Reply by MrXandbadas** [reply]
My git did not like the git clone command quoted so I replaced it with the one in the Readme.txt (for anyone trying this themselves)

BUT
You are an absolute GEM! Thanks!!

**Reply by taiya** [reply]
Yah, sorry, we are supposed to use SSH authentication at Google – forgot to switch back :)
Let me fix it above
